src/cmd/troubleshoot/check_dynakube.go

```diff
@@ -0,0 +1,268 @@
+			tslog.Errorf("'%s:%s' Dynakube isn't valid", troubleshootContext.namespaceName, troubleshootContext.dynakubeName)
+			return err
+		}
+	}
+
+	tslog.Okf("'%s:%s' Dynakube is valid", troubleshootContext.namespaceName, troubleshootContext.dynakubeName)
+	return nil
+}
+
+func checkDynakubeCrdExists(apiReader client.Reader, troubleshootContext *TestData) error {
+	dynakubeList := &dynatracev1beta1.DynaKubeList{}
+	if err := apiReader.List(context.TODO(), dynakubeList, &client.ListOptions{Namespace: troubleshootContext.namespaceName}); err != nil {
+		tslog.WithErrorf(err, "CRD for Dynakube missing")
+		return err
+	}
+	tslog.Infof("CRD for Dynakube exists")
+	return nil
+}
+
+func checkSelectedDynakubeExists(apiReader client.Reader, troubleshootContext *TestData) error {
+	dynakube := dynatracev1beta1.DynaKube{}
+	if err := apiReader.Get(context.TODO(), client.ObjectKey{Name: troubleshootContext.dynakubeName, Namespace: troubleshootContext.namespaceName}, &dynakube); err != nil {
+		tslog.WithErrorf(err, "selected '%s:%s' Dynakube does not exist", troubleshootContext.namespaceName, troubleshootContext.dynakubeName)
+		return err
+	}
+	tslog.Infof("using '%s:%s' Dynakube", troubleshootContext.namespaceName, troubleshootContext.dynakubeName)
+	return nil
+}
+
+func checkApiUrl(apiReader client.Reader, troubleshootContext *TestData) error {
+	tslog.Infof("checking if api url is valid...")
+
+	dynakube := dynatracev1beta1.DynaKube{}
+	if err := apiReader.Get(context.TODO(), client.ObjectKey{Name: troubleshootContext.dynakubeName, Namespace: troubleshootContext.namespaceName}, &dynakube); err != nil {
+		tslog.WithErrorf(err, "selected '%s:%s' Dynakube does not exist", troubleshootContext.namespaceName, troubleshootContext.dynakubeName)
+		return err
+	}
+
+	apiUrl := dynakube.Spec.APIURL
+
+	if len(apiUrl) < 4 || apiUrl[len(apiUrl)-4:] != "/api" {
+		tslog.Errorf("api url has to end on '/api' (%s)", apiUrl)
+		return fmt.Errorf("api url has to end on '/api' (%s)", apiUrl)
+	}
+
+	tslog.Infof("api url correctly ends on '/api'")
+
+	tslog.Infof("api url is valid")
+	return nil
+}
+
+func checkDynatraceApiSecret(apiReader client.Reader, troubleshootContext *TestData) error {
+	tslog.Infof("checking if secret is valid ...")
+
+	// use dynakube name or tokens value if set
+	troubleshootContext.dynatraceApiSecretName = troubleshootContext.dynakubeName
+
+	dynakube := dynatracev1beta1.DynaKube{}
+	if err := apiReader.Get(context.TODO(), client.ObjectKey{Name: troubleshootContext.dynakubeName, Namespace: troubleshootContext.namespaceName}, &dynakube); err != nil {
+		tslog.WithErrorf(err, "selected '%s:%s' Dynakube does not exist", troubleshootContext.namespaceName, troubleshootContext.dynakubeName)
+		return err
+	}
+
+	if dynakube.Spec.Tokens != "" {
+		troubleshootContext.dynatraceApiSecretName = dynakube.Spec.Tokens
+	}
+	return nil
+}
+
+func checkIfDynatraceApiSecretWithTheGivenNameExists(apiReader client.Reader, troubleshootContext *TestData) error {
```

Comment:
most functions have ,`troubleshootContext *TestData` as parameter, it would make sense to have pointer receivers instead